kernel/yosys.h

```diff
@@ -244,7 +244,7 @@ extern bool memhasher_active;
 inline void memhasher() { if (memhasher_active) memhasher_do(); }
 
 void yosys_banner();
-int ceil_log2(int x);
+int ceil_log2(const int x) YS_ATTRIBUTE(const) YS_ATTRIBUTE(pure);
```

Comment:
From the [GCC manual](https://embarc.org/man-pages/gcc/Common-Function-Attributes.html):
> Decorating the same function with both the const and the pure attribute is diagnosed.

So a function is either `pure` or `const`, but it can't have both attributes. (This function is `const` because it is not reading any global memory. If it would read global memory it would only be `pure`.)

Reply:
Removed the `pure` annotation